### PR TITLE
Interpret etcd not found error on a update

### LINKED ISSUE #18558: e2e flake: Kubectl client Kubectl run job should create a job from an image when restart is OnFailure [Conformance]
```
Kubernetes e2e suite.Kubectl client Kubectl run job should create a job from an image when restart is OnFailure [Conformance]
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:822
Expected error:
    <*errors.errorString | 0xc20853a8f0>: {
        s: "Error running &{/jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace@2/kubernetes/platforms/linux/amd64/kubectl [kubectl --server=https://130.211.162.64 --kubeconfig=/var/lib/jenkins/jobs/kubernetes-pull-build-test-e2e-gce/workspace@2/.kube/config stop jobs e2e-test-nginx-job --namespace=e2e-tests-kubectl-xhzko] []  <nil>  Command \"stop\" is deprecated, use \"delete\" instead.\nError from server: pods \"e2e-test-nginx-job-jrbq0\" not found\n [] <nil> 0xc2084e7be0 exit status 1 <nil> true [0xc2083f29b8 0xc2083f29e0 0xc2083f2a00] [0xc2083f29b8 0xc2083f29e0 0xc2083f2a00] [0xc2083f29d0 0xc2083f29f8] [0x817c10 0x817c10] 0xc208371380}:\nCommand stdout:\n\nstderr:\nCommand \"stop\" is deprecated, use \"delete\" instead.\nError from server: pods \"e2e-test-nginx-job-jrbq0\" not found\n\n",
    }
    Error running &{/jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace@2/kubernetes/platforms/linux/amd64/kubectl [kubectl --server=https://130.211.162.64 --kubeconfig=/var/lib/jenkins/jobs/kubernetes-pull-build-test-e2e-gce/workspace@2/.kube/config stop jobs e2e-test-nginx-job --namespace=e2e-tests-kubectl-xhzko] []  <nil>  Command "stop" is deprecated, use "delete" instead.
    Error from server: pods "e2e-test-nginx-job-jrbq0" not found
     [] <nil> 0xc2084e7be0 exit status 1 <nil> true [0xc2083f29b8 0xc2083f29e0 0xc2083f2a00] [0xc2083f29b8 0xc2083f29e0 0xc2083f2a00] [0xc2083f29d0 0xc2083f29f8] [0x817c10 0x817c10] 0xc208371380}:
    Command stdout:
    
    stderr:
    Command "stop" is deprecated, use "delete" instead.
    Error from server: pods "e2e-test-nginx-job-jrbq0" not found
    
    
not to have occurred
```
http://kubekins.dls.corp.google.com:8081/job/kubernetes-pull-build-test-e2e-gce/20148/#

### PR DESCRIPTION
**Pull request by janetkuo**
Fixes #18558

@bgrant0607 @kubernetes/sig-api-machinery @soltysh @wojtek-t

### CODE REVIEW

File: pkg/api/errors/etcd/etcd.go [hunk 1]
```diff
@@ -69,6 +69,8 @@ func InterpretUpdateError(err error, qualifiedResource unversioned.GroupResource
 		return errors.NewConflict(qualifiedResource, name, err)
 	case storage.IsUnreachable(err):
 		return errors.NewServerTimeout(qualifiedResource, "update", 2) // TODO: make configurable or handled at a higher level
+	case storage.IsNotFound(err):
```
Comment (janetkuo) [inline pkg/api/errors/etcd/etcd.go]: @smarterclayton is there any specific reason we didn't handle not found on etcd update?

**Comment by smarterclayton** [comment]
Lgtm

**Comment by soltysh** [comment]
LGTM

**Comment by soltysh** [comment]
@janetkuo thanks for nailing that one down!

**Comment by lavalamp** [comment]
LGTM

**Comment by wojtek-t** [comment]
LGTM - thanks @janetkuo

**Comment by eparis** [comment]
This PR is already included in the release-1.2 branch and does not require cherrypick-ing

**Pull request merged**